### PR TITLE
Adding max_area to random_size_crop

### PR DESCRIPTION
**Pull request by hetong007**
## Description ##
Enabling `random_size_crop` to set maximum area to crop for better data augmentation.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [X] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [X] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: python/mxnet/image/image.py [hunk 1]
```diff
@@ -457,9 +458,12 @@ def random_size_crop(src, size, min_area, ratio, interp=2):
 
     """
     h, w, _ = src.shape
-    area = h * w
+    src_area = h * w
+    if isinstance(area, float):
+        area = max(min(area, 1.0), 0.0)
```
Comment (piiswrong) [inline python/mxnet/image/image.py]: no need for this.

File: python/mxnet/image/image.py [hunk 2]
```diff
@@ -457,9 +458,12 @@ def random_size_crop(src, size, min_area, ratio, interp=2):
 
     """
     h, w, _ = src.shape
-    area = h * w
+    src_area = h * w
+    if isinstance(area, float):
```
Comment (piiswrong) [inline python/mxnet/image/image.py]: check for base.numeric_types

File: python/mxnet/image/image.py [hunk 3]
```diff
@@ -596,7 +600,7 @@ class RandomSizedCropAug(Augmenter):
     ----------
     size : tuple of (int, int)
         Size of the crop formatted as (width, height).
-    min_area : int
+    min_area : float in (0, 1]
```
Comment (piiswrong) [inline python/mxnet/image/image.py]: this doc and argument name needs to be changed too?

File: docs/tutorials/python/types_of_data_augmentation.md [hunk 4]
```diff
@@ -302,7 +302,7 @@ Some shortcut functions are provided to perform multiple augmentation in a singl
 ```python
 # A random crop, with a random resizing, and random aspect ratio jitter
 example_image_copy = example_image.copy()
-aug = mx.image.RandomSizedCropAug(size=(100, 100), min_area=0.1, ratio=(1.0, 1.5))
+aug = mx.image.RandomSizedCropAug(size=(100, 100), area=0.1, ratio=(1.0, 1.5))
```
Comment (piiswrong) [inline docs/tutorials/python/types_of_data_augmentation.md]: So looks like people are already using min_area as keyword argument.
We should keep backward compatibility by adding **kwargs to RandomSizedCropAug and random_size_crop and check if min_area is present, then issue a warning and override area

Comment (hetong007) [reply docs/tutorials/python/types_of_data_augmentation.md]: Do we have a standard warning format for deprecated parameters?

Comment (piiswrong) [reply docs/tutorials/python/types_of_data_augmentation.md]: no. somehting like xx is deprecated and renamed to xxx

File: python/mxnet/image/image.py [hunk 5]
```diff
@@ -457,9 +459,15 @@ def random_size_crop(src, size, min_area, ratio, interp=2):
 
     """
     h, w, _ = src.shape
-    area = h * w
+    src_area = h * w
+
+    if 'min_area' in kwargs:
+        warnings.warn('`min_area` is deprecated. Please use `area` instead.', DeprecationWarning)
+        area = kwargs.get('min_area')
```
Comment (piiswrong) [inline python/mxnet/image/image.py]:
```
if 'min_area' in kwargs:
        warnings.warn('`min_area` is deprecated. Please use `area` instead.', DeprecationWarning)
        area = kwargs.pop('min_area')
assert not kwargs, "unexpected keyword arguments ..."
```

**Pull request merged**